src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -154,28 +155,54 @@ public override async ValueTask<FormulaValue> Visit(LazyEvalNode node, EvalVisit
-        private async Task<FormulaValue> TryHandleSet(CallNode node, EvalVisitorContext context)
-        {
-            // Special case Set() calls because they take an LValue. 
-            if (node.Function.GetType() != typeof(RecalcEngineSetFunction))
-            {
-                return null;
-            }
-
-            var arg0 = node.Args[0];
-            var arg1 = node.Args[1];
-
+        private async Task<FormulaValue> TryHandleSet(CallNode node, EvalVisitorContext context)
+        {
+            // Special case Set() calls because they take an LValue. 
+            if (node.Function.GetType() != typeof(RecalcEngineSetFunction))
+            {
+                return null;
+            }
+
+            var arg0 = node.Args[0];
+            var arg1 = node.Args[1];
+
             var newValue = await arg1.Accept(this, context).ConfigureAwait(false);
 
-            if (arg0.IRContext.IsMutation && arg0 is RecordFieldAccessNode rfan)
+            if (arg0.IRContext.IsMutation)
             {
-                var arg0value = await rfan.From.Accept(this, context).ConfigureAwait(false);
-                
-                if (arg0value is RecordValue rv)
+                if (arg0 is RecordFieldAccessNode rfan)
                 {
-                    rv.ShallowCopyFieldInPlace(rfan.Field);
-                    rv.UpdateField(rfan.Field, newValue);
-                    return node.IRContext.ResultType._type.Kind == DKind.Boolean ? FormulaValue.New(true) : FormulaValue.NewVoid();
+                    var arg0value = await rfan.From.Accept(this, context).ConfigureAwait(false);
+
+                    if (arg0value is RecordValue rv)
+                    {
+                        rv.ShallowCopyFieldInPlace(rfan.Field);
+                        rv.UpdateField(rfan.Field, newValue);
+                        return node.IRContext.ResultType._type.Kind == DKind.Boolean ? FormulaValue.New(true) : FormulaValue.NewVoid();
+                    }
+                    else
+                    {
+                        return CommonErrors.UnreachableCodeError(node.IRContext);
+                    }
+                }
+                else if (arg0 is BinaryOpNode bon && bon.Op == BinaryOpKind.DynamicGetField)
+                {
+                    var arg0value = await bon.Left.Accept(this, context).ConfigureAwait(false);
+
+                    if (arg0value is UntypedObjectValue uov && uov.Impl is UntypedObjectBase impl)
+                    {
+                        TextLiteralNode textLiteralNode = (TextLiteralNode)bon.Right;
+
+                        try
+                        {
+                            impl.SetProperty(textLiteralNode.LiteralValue, newValue);
+                            return node.IRContext.ResultType._type.Kind == DKind.Boolean ? FormulaValue.New(true) : FormulaValue.NewVoid();
+                        }
+                        catch (CustomFunctionErrorException ex)

```

Comment:
>catch (CustomFunctionErrorException ex) [](http://example.com/codeflow?start=24&length=39)

since default impl is NotImplementedException, can we also catch that?